mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -397,10 +432,12 @@ impl NexusChild {
             });
         }
 
-        // Dropping the last descriptor results in the bdev being removed.
-        // This must be performed in this function.
-        let desc = self.desc.take();
-        drop(desc);
+        if destroying {
+            // Dropping the last descriptor results in the bdev being removed.
+            // This must be performed in this function.
+            let desc = self.desc.take();
+            drop(desc);
+        }
```

Comment:
I suspect the drop is now redundant as desc is about to go out of scope.

Reply:
Well spotted, I've removed the drop.